session/pingpong/accountant_promise_handler_test.go

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pingpong
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/mysteriumnetwork/node/eventbus"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/payments/crypto"
+	"github.com/rs/zerolog/log"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestAccountantPromiseHandler_RequestPromise(t *testing.T) {
+	aph := &AccountantPromiseHandler{
+		deps: AccountantPromiseHandlerDeps{
+			AccountantCaller:         &mockAccountantCaller{},
+			Encryption:               &mockEncryptor{},
+			EventBus:                 eventbus.New(),
+			AccountantPromiseStorage: &mockAccountantPromiseStorage{},
+		},
+		queue: make(chan enqueuedRequest),
+		stop:  make(chan struct{}),
+	}
+	go aph.handleRequests()
```

Comment:
It could be better to test via public methods only (if possible) when testing such full flow.